frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -4095,7 +4095,9 @@ static const Type* resolveBuiltinTypeCtor(Context* context,
     auto second = ci.actual(1).type();
     if (first.isParam() && first.type()->isIntType() &&
         second.isType()) {
-      return TupleType::getStarTuple(context, first, second);
+      auto num = first.param()->toIntParam()->value();
+      std::vector<const Type*> eltTypes(num, second.type());
+      return TupleType::getValueTuple(context, eltTypes);
```

Comment:
This is the only place in the compiler -- as far as I can tell -- that resolves an expression like `N * t`, aka "star tuple". But we're now moving it away from using `getStarTuple`. This seems concerning to me. Could you explain why we can't use `getStarTuple`, and if we can't, whether we should remove that helper altogether?

Reply:
I think ``getStarTuple`` should be renamed to ``getVarArgTuple``, which I will go ahead and take care of.

Not sure how this came about originally, though it's 99.99% likely to be my fault.

Reply:
Looks like some logic/assumptions got baked in over time, making things confusing. ``getStarTuple`` mainly supported the desired behavior for when we're resolving varargs initially, without knowing substitutions for instantiation. I'm guessing it was erroneously used (by me) in the ``N * t`` case based on the name of the method. I've renamed it to ``getVarArgTuple``.

While there, I noticed that ``getQualifiedTuple`` was always setting ``isVarArgTuple`` to ``true``, which seemed wrong. I've added in an optional argument that gets used in the proper places.